src/3p-frame.js

```diff
@@ -256,6 +268,22 @@ function getCustomBootstrapBaseUrl(parentWindow, opt_strictForUnitTest) {
 }
 
 /**
+ * Returns a randomized sentinel value for 3p iframes.
+ * The format is "%d-%d" with the first value being the depth of current
+ * window in the window hierarchy and the second a random integer.
+ * @param {!Window} parentWindow
+ * @return {string}
+ * @visibleForTesting
+ */
+export function generateSentinel(parentWindow) {
+  let windowDepth = 0;
+  for (let win = parentWindow; win != window.top; win = win.parent) {
```

Comment:
Right, I switched to a consistent pattern of navigating the window hierarchy using win!=win.parent to identify the top of the hierarchy.